VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Private/InitialShape.cpp

```diff
@@ -325,19 +334,24 @@ UStaticMesh* CreateStaticMeshFromInitialShapePolygon(const FInitialShapePolygon&
 	TArray<const FMeshDescription*> MeshDescriptions;
 	MeshDescriptions.Emplace(&MeshDescription);
 
-	UStaticMesh* StaticMesh;
+	auto Name = MakeUniqueObjectName(GetTransientPackage(), UStaticMesh::StaticClass(), TEXT("InitialShape"));
+	UStaticMesh* StaticMesh = NewObject<UStaticMesh>(GetTransientPackage(), Name, RF_Public | RF_Standalone);
+
 #if WITH_EDITOR
-	const FString InitialShapeName = TEXT("InitialShape");
-	const FString PackageName = TEXT("/Game/Vitruvio/") + InitialShapeName;
-	UPackage* Package = CreatePackage(*PackageName);
-	const FName StaticMeshName = MakeUniqueObjectName(Package, UStaticMesh::StaticClass(), FName(InitialShapeName));
+	StaticMesh->SetNumSourceModels(MeshDescriptions.Num());
+#endif
 
-	StaticMesh = NewObject<UStaticMesh>(Package, StaticMeshName, RF_Public | RF_Standalone | RF_Transactional);
-#else
-	StaticMesh = NewObject<UStaticMesh>();
+	UStaticMesh::FBuildMeshDescriptionsParams Params;
+	Params.bCommitMeshDescription = true;
+	Params.bMarkPackageDirty = false;
+#if !WITH_EDITOR
+	Params.bFastBuild = true;

```

Comment:
code repetition, it would make sense to add a helper function to get the desired parms.